File: app/src/main/java/android/yushenko/openweather/data/DataLab.java
package android.yushenko.openweather.data;

import android.yushenko.openweather.model.Daily;
import android.yushenko.openweather.model.Hourly;

import java.util.ArrayList;
import java.util.List;

public class DataLab {

    private static DataLab sDataLab;
    private List<Hourly> mHourlyList;
    private List<Daily> mDailyList;

    public static DataLab get() {
        if (sDataLab == null) {
            sDataLab = new DataLab();
        }
        return sDataLab;
    }

    private DataLab() {
        mHourlyList = new ArrayList<>();
        mDailyList = new ArrayList<>();
    }

    public void addHourlyList(List<Hourly> list) {
        mHourlyList = list;
    }

    public List<Hourly> getHourlyList() {
        return mHourlyList;
    }

    public void addDailyList(List<Daily> list) {
        mDailyList = list;
    }

    public List<Daily> getDailyList() {
        return mDailyList;
    }
}

File: app/src/main/java/android/yushenko/openweather/data/DataLab.kt
package android.yushenko.openweather.data

import android.yushenko.openweather.model.Daily
import android.yushenko.openweather.model.Hourly
import java.util.*

class DataLab private constructor() {
    var hourlyList: List<Hourly>
        private set
    var dailyList: List<Daily>
        private set

    fun addHourlyList(list: List<Hourly>?) {
        hourlyList = list!!
    }

    fun addDailyList(list: List<Daily>?) {
        dailyList = list!!
    }

    companion object {
        private var sDataLab: DataLab? = null
        fun get(): DataLab? {
            if (sDataLab == null) {
                sDataLab = DataLab()
            }
            return sDataLab
        }
    }

    init {
        hourlyList = ArrayList()
        dailyList = ArrayList()
    }
}